packages/features/ee/teams/lib/payments.ts

```diff
@@ -95,8 +102,22 @@ export const updateQuantitySubscriptionFromStripe = async (teamId: number) => {
     if (!url) return;
     const team = await getTeamWithPaymentMetadata(teamId);
     const { subscriptionId, subscriptionItemId } = team.metadata;
+    const membershipCount = team.members.length;
+    const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+    const subscriptionQuantity = subscription.items.data.find(
+      (sub) => sub.id === subscriptionItemId
+    )?.quantity;
+    if (!subscriptionQuantity) throw new Error("Subscription not found");
+
+    if (membershipCount < subscriptionQuantity) {
+      console.info(`Team ${teamId} has less members than seats, skipping updating subscription.`);
+      return;
+    }
+
+    const newQuantity = membershipCount - subscriptionQuantity;
+
     await stripe.subscriptions.update(subscriptionId, {
-      items: [{ quantity: team.members.length, id: subscriptionItemId }],
+      items: [{ quantity: membershipCount + newQuantity, id: subscriptionItemId }],
```

Comment:
If an org had less than 30 members and new members were added, it would override the 30 minimum.